add $environments arguments to ping methods

johanrosenson opened the pull request:
## Status
READY

## Description
Adds a second argument to the `thenPingHoneybadger` and `pingHoneybadgerOnSuccess` methods that allows you to specify one or more environments when to checkin, it can be omitted to always checkin (current behaviour).

The PR includes tests.

I also added a `string` typehint for the $id argument, which should be 100% backwards compatible since the Reporter checkin method already does the same typehint.

## Examples
```php
$schedule->command(SendEmails::class)
        ->daily()
        ->thenPingHoneybadger('Jiy63Xw', 'production'); // will only checkin on production
```

```php
$schedule->command(SendEmails::class)
        ->daily()
        ->thenPingHoneybadger('Jiy63Xw', ['production', 'development']); // will only checkin on production and development
```

johanrosenson commented:
I tried adding tests for pingHoneybadgerOnSuccess but that was not possible, maybe this is the reason there was no tests for it already?

Anyway, this is because before Laravel 8.0.0 [the exitCode was not set properly](https://github.com/laravel/framework/pull/34420) for closure based schedules, and before 8.6.0 the [exitCode was not set properly](https://github.com/laravel/framework/pull/34420), so this makes it impossible to write a test for this that doesn't fall with `--prefer-lowest`, therefor i have removed the tests for now. The onSuccess actually works with earlier versions, as long as it's not a callback as in the tests.

I can the tests again, but i think need some guidance how you want them done in that case.
Or we put it of for the future until minimum version of Laravel is 8.6 :)

shalvah commented:
You can try including the tests and skipping them on lower versions. Something like:

```php
if (version_compare(app()->version(), '8.6.0', '<')) {
    $this->markTestSkipped("Laravel < 8.6 doesn't set proper return codes.");
    return;
}
```

johanrosenson commented:
Thanks @shalvah , i have made the change in that fashion now, and I also added a bonus test for `pingHoneybadgerOnSuccess` similar to the original test for `thenPingHoneybadger`.

I also refactored the checks inside the macros to try and satisfy codeclimate, i think the first version i had was easier to read and the new version didn't make codeclimate happy anyway. 
Do you always follow the advice given by codeclimate or only when reasonable? I think the code will be really messy if we try to refactor here if the only intention is to _trick it_.

shalvah reviewed (approved):
I like!👍

shalvah commented:
>Do you always follow the advice given by codeclimate or only when reasonable? I think the code will be really messy if we try to refactor here if the only intention is to trick it.

Eh, I agree. I don't believe in coding to the tool also. This works for me. Explicitness > unnecessary refactoring.

johanrosenson commented:
@shalvah thanks for the feedback :)
